surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,330 @@
+- On CI (Continuous Integration) infrastructure
+- On a remote or cloud server
+- Other (open response)
+
+### Which development workflows limit your productivity?
+
+Please rate how much do you struggle with compiler performance in the following development workflows.
+
+Type: matrix (optional)
+
+Workflows:
+
+- Type checking (change code, type check using *cargo check* or IDE)
+- Running tests (change code, run tests *cargo test* or IDE)
+- Unoptimized rebuilds (change code, rebuild without optimizations)
+- Optimized rebuilds (change code, rebuild with optimizations)
+- Workspace rebuilds (change a crate which causes multiple other crates in your workspace to be rebuilt)
+- Clean unoptimized builds (build a crate graph from scratch)
+- Clean optimized builds (build a crate graph from scratch)
+- CI (Continuous Integration) builds
+- Docker builds
+
+Options:
+
+- Big problem for me
+- Could be improved, but does not limit me
+- Not an issue for me at all
+- I don't use this workflow
+
+### Do you have any other Rust development workflows that you would like to mention?
+
+Type: free form (optional)
+
+### How large is the Rust project that you work on in terms of lines of Rust code?
+
+Please provide an answer for the Rust project where you most struggle with compilation times. Only include the lines of Rust code of your project, not of its dependencies. You can count the lines of code e.g. using [tokei](https://github.com/XAMPPRocky/tokei).
+
+Type: select one (optional)
+
+- Less than 2 thousand lines
+- 2-10 thousand lines
+- 11-50 thousand lines
+- 51-100 thousand lines
+- 101-500 thousand lines
+- More than 500 thousand lines of code
+
+### How large is the Rust project that you work on in terms of (Cargo) dependencies?
+
+Please provide an answer for the Rust project where you most struggle with compilation times. Include the number of external/third-party dependencies/crates (including transitive ones) compiled when you run e.g. *cargo build*.
+
+Type: select one (optional)
+
+- No dependencies
+- 1-10 dependencies
+- 11-50 dependencies
+- 51-100 dependencies
+- 101-200 dependencies
+- 201-300 dependencies
+- 301-500 dependencies
+- More than 500 dependencies
+
+### How long do you need to wait for the compiler to rebuild your code after making a change?
+
+Please select the longest duration range amongst the projects that you regularly work on.
+
+This question focuses on local development workflows, not CI or remote builds. If you do not compile your Rust code locally, please skip this question.
+
+Type: select one (optional)
+
+- Less than a second
+- Between 1 and 5 seconds
+- Between 5 and 10 seconds
+- Between 10 and 30 seconds
+- Between 30 seconds and 1 minute
+- Between 1 and 5 minutes
+- More than 5 minutes
+
+### Which of the following problems do you most struggle with?
+
+Type: matrix (optional)
+
+Commands:
+
+- Waiting for a rebuild after making a small change
+- Waiting for CI workflows that build Rust code
```

Comment:
I would like to suggest the following questions about CI:
- How long does the compiler take to build your project in CI?
- Does your CI cache previous build artifacts? If yes, what do you use for caching?